src/blenderbim/blenderbim/bim/module/spreadsheet_writer/operator.py

```diff
@@ -0,0 +1,652 @@
+                
+                return {'FINISHED'}
+
+                
+            if blenderbim_spreadsheet_properties.my_file_path.endswith(".ods"):
+                print ("Retrieving data from: " , blenderbim_spreadsheet_properties.my_file_path)
+            
+                # Get content xml data from OpenDocument file
+                ziparchive = zipfile.ZipFile(blenderbim_spreadsheet_properties.my_file_path, "r")
+                xmldata = ziparchive.read("content.xml")
+                ziparchive.close()
+                
+                # Create parser and parsehandler
+                parser = xml.parsers.expat.ParserCreate()
+                treebuilder = TreeBuilder()
+                # Assign the handler functions
+                parser.StartElementHandler  = treebuilder.start_element
+                parser.EndElementHandler    = treebuilder.end_element
+                parser.CharacterDataHandler = treebuilder.char_data
+
+                # Parse the data
+                parser.Parse(xmldata, True)
+
+                hidden_rows = get_hidden_rows(treebuilder.root)  # This returns a generator object
+        
+                dataframe = pd.read_excel(blenderbim_spreadsheet_properties.my_file_path, sheet_name=blenderbim_spreadsheet_properties.my_workbook, skiprows=hidden_rows, engine="odf")
+                self.filter_IFC_elements(context, guid_list=dataframe['GlobalId'].values.tolist())
+                
+                return {'FINISHED'}
+                
+                
+    def filter_IFC_elements(self, context, guid_list):
+        
+        print ("Filtering IFC elements")
+        
+        start_time = time.perf_counter()
+        
+        outliner = next(a for a in bpy.context.screen.areas if a.type == "OUTLINER") 
+        outliner.spaces[0].show_restrict_column_viewport = not outliner.spaces[0].show_restrict_column_viewport
+        
+        bpy.ops.object.select_all(action='DESELECT')
+      
+        for obj in bpy.context.view_layer.objects:
+            element = tool.Ifc.get_entity(obj)
+            if element is None:        
+                obj.hide_viewport = True
+                continue
+            data = element.get_info()
+       
+            
+            obj.hide_viewport = data.get("GlobalId", False) not in guid_list
+
+        bpy.ops.object.select_all(action='SELECT') 
+        
+        print (time.perf_counter() - start_time, "seconds to show the IFC elements")
+            
+
+        
+        
+    
+class UnhideIFCElements(bpy.types.Operator):
+    """Unhide all IFC elements"""
+    bl_idname = "object.unhide_all"
+    bl_label = "Unhide All"
+
+    def execute(self, context):
+        print("Unhide all")
+        
+        for obj in bpy.data.objects:
+            obj.hide_viewport = False 
+        
+        return {'FINISHED'}  
+ 
+ 
+class MyCollectionActions(bpy.types.Operator):
+    bl_idname = "my.collection_actions"
```

Comment:
Since this will be merged in a rather extensive codebase, it's a good idea to try and reduce the chance of having name collisions. You can rename this to something more specific eg. "bim.spreadsheet_collection_action"